src/ResourceManager/Insights/Commands.Insights/Diagnostics/SetAzureRmDiagnosticSettingCommand.cs

```diff
@@ -41,6 +41,27 @@ public class SetAzureRmDiagnosticSettingCommand : ManagementCmdletBase
         public string ResourceId { get; set; }
 
         /// <summary>
+        /// Gets or sets the switch for storage parameter of the cmdlet
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "Disable storage")]
+        [ValidateNotNullOrEmpty]
+        public SwitchParameter DisableStorage { get; set; }
+
+        /// <summary>
+        /// Gets or sets the switch for eventhub parameter of the cmdlet
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "Disable service bus")]
+        [ValidateNotNullOrEmpty]
+        public SwitchParameter DisableServiceBus { get; set; }
+
+        /// <summary>
+        /// Gets or sets the switch for workspace parameter of the cmdlet
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "Disable workspace")]
+        [ValidateNotNullOrEmpty]
+        public SwitchParameter DisableWorkspace { get; set; }
```

Comment:
The powershell is supposed to offer a way to easily build the request. If the request ends up not being valid, we want to the server to validate and respond with proper message.